packages/ipfs/src/core/ipns/publisher.js

```diff
@@ -44,11 +63,13 @@ class IpnsPublisher {
       throw errcode(new Error(errMsg), 'ERR_INVALID_PEER_ID')
     }
 
+    // @ts-ignore - Type defs for PeerId hav no _pubKey
     const publicKey = peerId._pubKey
     const embedPublicKeyRecord = await ipns.embedPublicKey(publicKey, record)
+    /** @type {any} */
     const keys = ipns.getIdKeys(peerId.toBytes())
 
-    await this._publishEntry(keys.routingKey, embedPublicKeyRecord || record, peerId)
+    await this._publishEntry(keys.routingKey, embedPublicKeyRecord || record)
```

Comment:
`_publishEntry` below takes just two arguments. I changed this because TS is rightfully points that out.